### ISSUE TITLE
Getting Timeout Error

### ISSUE THREAD

**Issue by mudittuli** [fwouts/react-screenshot-test#234]
Trying to run the generate screenshot

`jest -c jest.screenshot.config.js -u`

but getting the error

```
● Test suite failed to run

    Timeout - Async callback was not invoked within the 5000ms timeout specified by jest.setTimeout.Error: Timeout - Async callback was not invoked within the 5000ms timeout specified by jest.setTimeout.
      at mapper (node_modules/jest-jasmine2/build/queueRunner.js:27:45)
Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        12.95s
Ran all test suites.
  console.error node_modules/jest-jasmine2/build/jasmine/Env.js:248
    Unhandled error

  console.error node_modules/jest-jasmine2/build/jasmine/Env.js:249
    Error: socket hang up
        at connResetException (internal/errors.js:570:14)
        at Socket.socketOnEnd (_http_client.js:440:23)
        at Socket.emit (events.js:215:7)
        at Socket.EventEmitter.emit (domain.js:475:20)
        at endReadableNT (_stream_readable.js:1184:12)
        at processTicksAndRejections (internal/process/task_queues.js:80:21)
```

**Comment by fwouts** [comment]
Hi @mudittuli, do you have any mocks set up inside `__mocks__` that might be interfering?

**Comment by davsucks** [comment]
I'm experiencing the same symptom actually. My component does some networking when it renders so I'm mocking that out, I also notice that `useEffect` never gets called so I'm not sure if that has something to do with it. Even when I spy on `useEffect` to just call the function its passed I don't see anything in the component executing after that line.

**Comment by fwouts** [comment]
@davsucks it's expected that `useEffect` wouldn't be called, because React Screenshot Test only does server-side rendering (no client-side rendering, it's just static HTML). Could that explain your issue?

**Comment by davsucks** [comment]
Possibly? I've spun up the app with `create-react-app` so I'm not sure how that affects the components being rendered server-side. I noticed that when I render just a dummy component (e.g. hello world in a div) the tests work as expected but when I render the component I'd like to test it just hangs.

**Comment by davsucks** [comment]
Actually I figured out what the problem was. We're using `react-router-dom` and I was rendering a Link component which requires a Router component as a parent, although the error was being swallowed and not reported which was making it harder to debug.

Thanks for the response though! I think it would be nice to have a helper function like `.withProvider()` or `.withParent()` to let users pass in a component which would be rendered as a parent of your child when `.run()` is called. Are you open to pull requests/is this something you've thought of?

**Comment by fwouts** [comment]
Nice catch @davsucks. Did you end up fixing this by wrapping the component manually?

The idea of `.withParent()` sounds neat. Happy to review a PR if you have an implementation in mind (I'm hoping it would be quite small).

**Comment by davsucks** [comment]
Unfortunately I had to abandon adding this to our repo, we're using material UI and our stylesheets are written in JavaScript so we're inserting styles with hooks, which didn't seem to play nicely with server-side rendering. I noticed in the create react app example you have the styles are just css files. 

That said I noticed that material UI does have SSR guide but I couldn't figure out a way to reconcile that approach with this testing library 😕

Also, I did fork the repo and try giving `withParent` a crack but wasnt too pleased with the result. I might try to revisit it later though and see if I can make it PR friendly
